restui/views/mappings.py

```diff
@@ -270,39 +268,37 @@ def put(self, request, pk):
         except MultipleObjectsReturned:
             raise Http404("Couldn't get unique status for {}".format(request.data['status']))
 
-#        for old_status in UeMappingStatus.objects.filter(mapping=mapping).filter(~Q(status=s.id)):
-#            old_status.delete()
-
         # If the mapping has already been assigned that status, update the timestamp,
         # otherwise create one from scratch
         try:
-#            mapping_status = UeMappingStatus.objects.get(mapping=mapping, status=s.id)
-            mapping_status = UeMappingStatus.objects.get(mapping=mapping)
+            mapping_status = UeMappingStatus.objects.filter(mapping=mapping).latest('time_stamp')
```

Comment:
I think you can more conveniently use "mapping.status_history..." based on the change you've made to the annotations models.